sqlglot/transforms.py

```diff
@@ -109,6 +111,16 @@ def eliminate_qualify(expression: exp.Expression) -> exp.Expression:
         select_candidates = exp.Window if expression.is_star else (exp.Window, exp.Column)
         for expr in qualify_filters.find_all(select_candidates):
             if isinstance(expr, exp.Window):
+                for column in expr.find_all(exp.Column):
+                    for select in expression.selects:
```

Comment:
We shouldn't be iterating over `selects` like this. Let's instead create a dict of `select.alias_or_name` -> `select` for all `expression.selects` and do a lookup with `column.name` as the key.

Reply:
thanks for your amazing knowledge!